core/string/translation.cpp

```diff
@@ -657,6 +657,28 @@ StringName TranslationServer::_get_message_from_translations(const StringName &p
 	return res;
 }
 
+String TranslationServer::_get_best_locale(const HashSet<Ref<Translation>> &p_translations) const {
+	// Look for best matching translation.
+	String best_locale = "en";
+	int best_score = 0;
+
+	for (const Ref<Translation> &E : p_translations) {
+		const Ref<Translation> &t = E;
+		ERR_FAIL_COND_V(t.is_null(), best_locale);
+		String l = t->get_locale();
+
+		int score = compare_locales(locale, l);
```

Comment:
`locale` here is determined by the environment. But editor locale should be from `interface/editor/editor_language`.

Maybe the target locale should be an argument?

Reply:
Editor locale is passed in the first translation, so this code is wrong.